src/OpenTelemetry.AutoInstrumentation/Configurations/ResourceConfigurator.cs

```diff
@@ -20,24 +20,28 @@ namespace OpenTelemetry.AutoInstrumentation.Configurations;
 
 internal static class ResourceConfigurator
 {
-    public static void Configure(ResourceBuilder resourceBuilder)
+    public static ResourceBuilder CreateResourceBuilder()
     {
-        var pluginManager = Instrumentation.PluginManager;
-
-        resourceBuilder
+        var resourceBuilder = ResourceBuilder
+            .CreateEmpty() // Don't use CreateDefault because it puts service name unknown by default.
+            .AddEnvironmentVariableDetector()
             .AddTelemetrySdk()
             .AddAttributes(new KeyValuePair<string, object>[] { new(Constants.Tracer.AutoInstrumentationVersionName, Constants.Tracer.Version) });
 
+        var pluginManager = Instrumentation.PluginManager;
         if (pluginManager != null)
         {
             resourceBuilder.InvokePlugins(pluginManager);
         }
-    }
 
-    public static ResourceBuilder CreateResourceBuilder()
-    {
-        var resourceBuilder = ResourceBuilder.CreateDefault();
-        Configure(resourceBuilder);
+        var resource = resourceBuilder.Build();
+        var serviceName = resource.Attributes.FirstOrDefault(a => a.Key == "service.name").Value as string;
+        if (serviceName?.StartsWith("unknown_service:") == true)
```

Comment:
Earlier this was checking the actual value of the service name since the default from the SDK was always set, so we had to check if it was being overwritten. Now, we create the resource builder ourselves so we just need to check for the presence of the key "service.name".

Reply:
I'm unsure about this, but, could you please check if `ResourceSemanticConventions.AttributeServiceName` can be used here? If it is not available leave it as it is.

Reply:
> I'm unsure about this, but, could you please check if `ResourceSemanticConventions.AttributeServiceName` can be used here? If it is not available leave it as it is.

It's inaccessible unfortunately.

Reply:
> Earlier this was checking the actual value of the service name since the default from the SDK was always set, so we had to check if it was being overwritten. Now, we create the resource builder ourselves so we just need to check for the presence of the key "service.name".

That's a good point and I agree, we should just make sure it's not set by the other builder methods after CreateEmpty()

Will update this.